### PR TITLE
[SeeKeR] Opt construction for Python 3.8

### PR DESCRIPTION
**Pull request by klshuster**
**Patch description**
Fixes issue in #4451 

The opt construction for sub-opts for each agent takes the final option string in the parser; for python < 3.8, this would *always* be the `--` option when a `-` was also present. In Python >=3.8, this seems to be non-deterministic.

The new logic ensures we get the long option every time.

**Testing steps**
Tested sample commands from #4451

### CODE REVIEW

File: projects/seeker/agents/seeker.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def add_additional_subagent_args(cls, parser: ParlaiParser) -> ParlaiParser:
         """
         additional_agent_parser = cls.get_additional_agent_args()
         for action in additional_agent_parser._actions:
-            key = action.option_strings[-1]
+            key = sorted(action.option_strings, key=lambda x: len(x))[-1]
```
Comment (mojtaba-komeili) [inline projects/seeker/agents/seeker.py:204]: nit: do you think using `max` makes it slightly shorter? You wouldn't need the `[-1]`.

Comment (klshuster) [reply projects/seeker/agents/seeker.py:204]: oh yeah good point

Comment (klshuster) [reply projects/seeker/agents/seeker.py:204]: since we're dealing with strings not numbers

**Review by stephenroller — COMMENTED** [review]
Max instead of sorted...

File: projects/seeker/agents/seeker.py (hunk 1 above)
Comment (mojtaba-komeili) [reply projects/seeker/agents/seeker.py:204]: I don't understand, the behavior shouldn't change between the max or sort here.

**Pull request merged**